server/index.js

```diff
@@ -481,43 +483,64 @@ function updateSessionState( session, setState=null ) {
-    serverData[ "sessions" ][ session ][ "players" ][ player ][ "spaceship" ][ "position" ] = [ 50, 50 ];
-    serverData[ "sessions" ][ session ][ "players" ][ player ][ "spaceship" ][ "angle" ] = 0;
-    serverData[ "sessions" ][ session ][ "players" ][ player ][ "spaceship" ][ "resources" ] = 0;
-    serverData[ "sessions" ][ session ][ "players" ][ player ][ "planet" ][ "position" ] = [ 50, 50 ];
-    serverData[ "sessions" ][ session ][ "players" ][ player ][ "planet" ][ "resources" ] = 0;
-  } );
+  Object.keys(serverData["sessions"][session]["players"]).forEach((player) => {
+    serverData["sessions"][session]["players"][player]["spaceship"]["position"] = [50, 50];
+    serverData["sessions"][session]["players"][player]["spaceship"]["angle"] = 0;
+    serverData["sessions"][session]["players"][player]["spaceship"]["resources"] = 0;
+    serverData["sessions"][session]["players"][player]["planet"]["position"] = [50, 50];
+    serverData["sessions"][session]["players"][player]["planet"]["resources"] = 0;
+    serverData["sessions"][session]["players"][player]["spaceship"]["isSpectating"] = false;
+  });
 
-  sendMessageToSession( "sessionReset", { "info": serverData[ "sessions" ][ session ] }, session );
+  sendMessageToSession("sessionReset", { "info": serverData["sessions"][session] }, session);
 }
 
-function resetSessionTime( session ) {
-  serverData[ "sessions" ][ session ][ "remainingTime" ] = serverData[ "sessions" ][ session ][ "time" ];
+function resetSessionTime(session) {
+  serverData["sessions"][session]["remainingTime"] = serverData["sessions"][session]["time"];
 }
 
 function updateTime() {
-  Object.keys( serverData[ "sessions" ] ).forEach( ( session ) => {
-    if ( serverData[ "sessions" ][ session ][ "state" ] == "playing" ) {
-      serverData[ "sessions" ][ session ][ "remainingTime" ] -= 1000;
-
-      sendMessageToSession( "timeUpdated", { "remainingTime": serverData[ "sessions" ][ session ][ "remainingTime" ] }, session );
-
-      updateSessionState( session )
+  Object.keys(serverData["sessions"]).forEach((session) => {
+    if (serverData["sessions"][session]["state"] == "playing") {
+      serverData["sessions"][session]["remainingTime"] -= 1000;
+
+      sendMessageToSession("timeUpdated", { "remainingTime": serverData["sessions"][session]["remainingTime"] }, session);
+
+      updateSessionState(session);
+
+      // Update spectating players' respawn timers
+      Object.keys(serverData["sessions"][session]["players"]).forEach((player) => {
+        let spaceship = serverData["sessions"][session]["players"][player]["spaceship"];
+
+        // Check to see which players are spectating (i.e. crashed)
+        if (spaceship['isSpectating']) {
+
+          // Spaceship["respawnTime"] is always null at the moment of the crash because players don't include it in their normal updates. See (action == "updateSpaceship") if clause.
```

Comment:
Same point I mentioned above, we can have the default respawnTime set to 0, so we don't have to do the null checks.
We're not using that value unless we're spectating anyway, and in which case we've already set the respawnTime to 5.